### PR TITLE
Add Title Search Functionality and Convert Type from JS to TS

### PR DESCRIPTION
**Pull request by SariusGH**
* Add `TitleCard.tsx` as new possible result card of the search bar
* Add functionality to make titles searchable
* Convert type `GlobalChallengeDto` from JS to `GlobalChallengeDTO` in TS

### CODE REVIEW

File: components/Search/Searchbar.tsx [hunk 1]
```diff
@@ -110,6 +131,36 @@ export default function Searchbar() {
...
             </>
             : null}
+         {titles && searchValue.length > 0 ?
+            <>
+               {titles.map((title) => {
+                  if (!title.name.toLowerCase().startsWith(searchValue)) {
+                     return null;
+                  }
+
+                  if (title.challengeId === undefined) { // TODO titles with no challange
```
Comment (DarkIntaqt) [inline components/Search/Searchbar.tsx:141]: this whole bracket can be removed, it is duplicate / no response should be default and therefore fallback here is not needed

File: components/Search/Searchbar.tsx [hunk 2]
```diff
@@ -110,6 +131,36 @@ export default function Searchbar() {
...
+                     return;
+                     return <TitleCard
+                     key={title.id}
+                     title={title.name}
+                     url={"/challenges/" + title.challengeId} // TODO think of a link
+                     image={contentService.getChallengeTokenIcon(parseInt("1"), "MASTER")} // TODO think of a icon
+                  />;
+                  }
+
+                  if (title.type === "CHALLENGE") {
```
Comment (DarkIntaqt) [inline components/Search/Searchbar.tsx:151]: type can be either `CHALLENGE` or `DEFAULT`

File: components/Search/TitleCard.tsx
Comment (DarkIntaqt) [inline components/Search/TitleCard.tsx:1]: It'd be nice to show a title icon or so, I can add that if you want to, you could add it as well :shrug:

**Review by DarkIntaqt — CHANGES_REQUESTED** [review]
Please change the stuff pointed out. The rewrite from js to ts is good

File: components/Search/Searchbar.tsx (hunk 2 above)
Comment (SariusGH) [reply components/Search/Searchbar.tsx]: Fixed

File: components/Search/Searchbar.tsx (hunk 1 above)
Comment (SariusGH) [reply components/Search/Searchbar.tsx]: Fixed in https://github.com/SariusGH/challenges/commit/f0be20bfbbdf41da709d39b03e684160292ab184

File: components/Search/TitleCard.tsx
Comment (SariusGH) [reply components/Search/TitleCard.tsx:1]: I currently don't have the available resources to add this icon. It's your turn now o7

**Review by DarkIntaqt — APPROVED** [review]
LGTM

**Pull request merged**